From b77fc64f0e986d4b72271b47a10496ba9d51c629 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 21 Jun 2018 11:38:55 +0800
Subject: [PATCH 4000/5242] MLK-18649-2 arm64: fsl-imx8dx.dtsi: Add
 pd_mipi_dsi_0/1_dual_lvds power domains

commit  89283eb55aae31cabe20a9577a77ceea2f512a3b from
https://source.codeaurora.org/external/imx/linux-imx.git

When ldb works in dual LVDS channel mode, it consumes two MIPI/LVDS combo
subsystems, one primary and the other auxiliary.  Each subsystem is powered
by it's own power domain.  So, the dual channel mode needs two power domains.
This patch adds pd_mipi_dsi_0/1_dual_lvds power domains support so that
the ldb1/2 nodes may use them to support dual channel mode.  Note that
the pd_mipi_dsi_0/1_aux_lvds are the parent power domains of the
pd_mipi_dsi_0/1_dual_lvds power domains respectively.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit 86c85ca8cf5b81d4e97e54955ea46e50fdb76b9f)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi |   28 +++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 51dcc10..bd43b4b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -1121,6 +1121,20 @@
 					power-domains =<&pd_mipi_dsi0>;
 				};
 
+				pd_mipi_dsi_0_aux_lvds: PD_AUX_LVDS0 {
+					reg = <SC_R_LVDS_0>;
+					#power-domain-cells = <0>;
+					power-domains = <&pd_mipi_dsi0>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					pd_mipi_dsi_1_dual_lvds: PD_DUAL_LVDS1 {
+						reg = <SC_R_LVDS_1>;
+						#power-domain-cells = <0>;
+						power-domains = <&pd_mipi_dsi_0_aux_lvds>;
+					};
+				};
+
 				pd_mipi_dsi_0_i2c0: PD_MIPI_0_DSI_I2C0 {
 					reg = <SC_R_MIPI_0_I2C_0>;
 					#power-domain-cells = <0>;
@@ -1151,6 +1165,20 @@
 					power-domains =<&pd_mipi_dsi1>;
 				};
 
+				pd_mipi_dsi_1_aux_lvds: PD_AUX_LVDS1 {
+					reg = <SC_R_LVDS_1>;
+					#power-domain-cells = <0>;
+					power-domains = <&pd_mipi_dsi1>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					pd_mipi_dsi_0_dual_lvds: PD_DUAL_LVDS0 {
+						reg = <SC_R_LVDS_0>;
+						#power-domain-cells = <0>;
+						power-domains = <&pd_mipi_dsi_1_aux_lvds>;
+					};
+				};
+
 				pd_mipi_dsi_1_i2c0: PD_MIPI_1_DSI_I2C0 {
 					reg = <SC_R_MIPI_1_I2C_0>;
 					#power-domain-cells = <0>;
-- 
1.7.9.5

